desktop/core/src/desktop/js/reactComponents/FileChooser/api.ts

```diff
@@ -0,0 +1,29 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { post } from '../../api/utils';
+import { CancellablePromise } from '../../api/cancellablePromise';
+
+const FILESYSTEMS_API_URL = '/api/storage/get_filesystems';
+
+export interface FetchFileSystemsResponse {
+  filesystems: Array<string>;
+  status: number;
+}
+
+export const fetchFileSystems = (): CancellablePromise<FetchFileSystemsResponse> => {
+  return post(FILESYSTEMS_API_URL);
```

Comment:
Why is this 'post' and not 'get'?

Reply:
The get_filesystems public api uses post. Please have a look at #3075 for reference.

Reply:
Good catch @bjornalm @nidhibhatg! I feel it makes no sense for having this as a POST API. I'll change it to accept only GET requests and update the documentation. 

Meanwhile, you can change it to a GET call.